src/preamble/01_what_do_we_mean_when_we_say_cabal.md

```diff
@@ -1 +1,97 @@
 # What do we mean when we say cabal?
+
+Cabal is an umbrella term that can refer to either "cabal the spec" (.cabal
+files), "cabal the library" (code that understands .cabal files), or "cabal the
+tool" (the cabal-install package which provides the cabal executable).
+
+## tl;dr
+
+1. [.cabal file](#the-cabal-file) This is the file that specifies the content of
+   Haskell packages. The types of information that can be provided via the
+   bespoke file format are: top level metadata about the package, and build
+   specific information (exposed modules, external dependencies, language
+   extensions, compiler options) about the components that comprise the package.
+
+2. [The Cabal build system](#the-cabal-library) is a library that is used for
+   creating packages and building their contents. This build system is used by
+   cabal-install and other package managers, stack for example.
+
+3. [cabal-install](#the-binary-cabal-install-cli-tool) is the command line
+   utility to help configure, compile and install Haskell libraries and
+   programs.
+
+## The .cabal file
+
+The .cabal file contains information that drives the compilation and building of
+Haskell packages. The .cabal file lives at the root directory that contains the
+haskell source code corresponding to the package. By convention this file is
+named `<project-name>.cabal`. It is a text-based, key-value format, that is
+divided into subsections called stanzas.
+
+The package properties describe the package as a whole, such as name, license,
+author, dependenices, etc. It also contains optional information about optional
+components such as
+[library properties](../new_to_cabal/06_first_cabal_library.md),
```

Comment:
Nitpick: 
"about optional components such as library properties" - sounds weird. Is "library property| an optional component?
I'd change this to `[libraries]`. Also in the "optional information about optional components" I'd drop the first "optional".
Since the component is optional, it means it can be left out, so no need to mention the optionality twice.

Reply:
Yep, agree. I'll remove the first optional.